fabric-fluids-v1/src/main/java/net/fabricmc/fabric/api/fluids/containers/volume/ImmutableFluidVolume.java

```diff
@@ -0,0 +1,90 @@
+package net.fabricmc.fabric.api.fluids.containers.volume;
+
+import java.util.Objects;
+
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.api.fluids.minecraft.FluidIds;
+
+public class ImmutableFluidVolume implements FluidVolume {
+	public static final ImmutableFluidVolume EMPTY = new ImmutableFluidVolume();
+
+	private final Identifier fluid;
+	private final long amount;
+	private final CompoundTag tag;
+
+	public ImmutableFluidVolume() {
+		this(FluidIds.EMPTY, 0);
+	}
+
+	public ImmutableFluidVolume(Identifier fluid, long amount) {
+		this(fluid, amount, new CompoundTag());
+	}
+
+	public ImmutableFluidVolume(Identifier fluid, long amount, CompoundTag tag) {
+		this.fluid = fluid;
+		this.amount = amount;
+		this.tag = tag;
+	}
+
+	@Override
+	public FluidVolume drain(Identifier fluid, long amount, boolean simulate) {
+		return EMPTY;
+	}
+
+	@Override
+	public FluidVolume add(FluidVolume container, boolean simulate) {
+		return EMPTY;
+	}
+
+	@Override
+	public boolean isEmpty() {
+		return false;
```

Comment:
Why is this class named `ImmutableFluidVolume`, with constant `EMPTY`, and always returning `false` for `isEmpty` when the volume is not always guaranteed to be 0?

Reply:
oops 😅